react/features/stream-effects/blur/JitsiStreamCropPersonEffect.js

```diff
@@ -0,0 +1,193 @@
+     *
+     * @private
+     * @param {EventHandler} response - The onmessage EventHandler parameter.
+     * @returns {void}
+     */
+    _onMaskFrameTimer(response) {
+        if (response.data.id === INTERVAL_TIMEOUT) {
+            this._renderMask();
+        }
+    }
+
+    /**
+     * Starts loop to capture video frame and render the segmentation mask.
+     *
+     * @param {MediaStream} stream - Stream to be used for processing.
+     * @returns {MediaStream} - The stream with the applied effect.
+     */
+    startEffect(desktopStream) {
+        
+        // get the stream properties for the screenshare
+        const desktopTrack = desktopStream.getVideoTracks()[0];
+        const { height, frameRate, width }
+            = desktopTrack.getSettings ? desktopTrack.getSettings()
+                : desktopTrack.getConstraints();
+        this._height = height;
+        this._width = width;
+        this._frameRate = frameRate;
+        this._inputDesktopElement.width = width;
+        this._inputDesktopElement.height = height;
+        this._inputDesktopElement.autoplay = true;
+        this._inputDesktopElement.srcObject = desktopStream;
+
+        const videoDiv = document.createElement("div");
+        document.body.appendChild(videoDiv);
+        videoDiv.appendChild(this._inputVideoElement);
+        videoDiv.appendChild(this._inputDesktopElement);
+        // set the style attribute of the div to make it invisible
+        videoDiv.style.display = "none";
+
+        this._maskCanvasElement.width = this._cameraWidth;
+        this._maskCanvasElement.height = this._cameraHeight;
+        this._maskCanvasContext = this._maskCanvasElement.getContext('2d');
+        this._outputCanvasElement.width = this._width;
+        this._outputCanvasElement.height = this._height;
+
+        this._videoFrameTimerWorker.postMessage({
+            id: SET_INTERVAL,
+            timeMs: 1000 / this._cameraFrameRate
+        });
+        this._maskFrameTimerWorker.postMessage({
+            id: SET_INTERVAL,
+            timeMs: 50
+        });
+
+        return this._outputCanvasElement.captureStream(this._cameraFrameRate);
+    }
+
+    /**
+     * Stops the capture and render loop.
+     *
+     * @returns {void}
+     */
+    stopEffect() {
+        this._videoFrameTimerWorker.postMessage({
+            id: CLEAR_INTERVAL
+        });
+        this._maskFrameTimerWorker.postMessage({
+            id: CLEAR_INTERVAL
+        });
+    }
+
+    /**
+     * Loop function to render the video frame input and draw blur effect.
+     *
+     * @private
+     * @returns {void}
+     */
+    _renderVideo() {
+        this._maskCanvasContext.drawImage(this._inputVideoElement, 0, 0,
+            this._cameraWidth, this._cameraHeight);
+        if (this._segmentationData) {
+            const img = this._maskCanvasContext.getImageData(0, 0, 
+                this._cameraWidth, this._cameraHeight);
```

Comment:
Is it weird for me to think the readability could be improved by using some JS classes to encapsulate some calculations? Maybe this is actually simple enough logic, but on first read I wonder if moving some of this logic into methods with english names would allow me to understand quicker the steps that are going on.

Reply:
Maybe better comments ? This is actually simple enough logic like you mentioned :)